Passing data to send to websocket

finga opened the pull request:
When using `WebSocketTask::send()` and `WebSocketTask::send_binary()` it feels rather strange to pass the data as a `Result` to the function. Would it not make more sense to pass `String` and `Vec<u8>` respectively?

Please let me know what you think about that :)

Commit:
When sending data pass the actual data

When sending data (binary as `Vec<u8>` or text as `String`) pass the
actual data instead of having it encapsulated in a `Result<(Vec<u8>,
anyhow::Error)>` or `Result<(String, anyhow::Error)>`.

Commit:
Remove passing as references

Remove passing as references as they are immediately dereferenced by
the compiler.

darioalessandro commented:
I agree, I took this pattern straight from the yew repo. 

Do we need to keep those Json macros? https://github.com/security-union/yew-websocket/blob/master/src/macros.rs

Is there any dead code after this?

finga commented:
Glad to hear, I will take a look if those macros are still needed and if there will be any dead code. I just wanted to wait for your point of view on this before digging any deeper. Further, I will adapt the example to reflect those changes as well. I will let you know here when I think I am done with everything. Hope that is fine for you.

One last thing, since this is a breaking change, when releasing this, I think an increase in the major version number is needed, but I am not entirely sure as I am not too familiar with semver.

darioalessandro commented:
sounds good, lets increase the major version, no big deal, it is early enough

Commit:
example: Adapt to send unwrapped data

As `WebSocketTask::send()` and `WebSocketTask::send_binary()` do not
expect a `Result` anymore, adapt the example to pass "just" the data.

Commit:
macros: Remove serialization

Due to the latest changes, serialization should not be needed anymore.

finga commented:
I pushed what I did with the example and the macros. I am not fully happy with it, as I wanted to get rid of the whole `Json` derive macro code, but I could not find a better solution for that. What do you think about it?

finga commented:
Just wanted to ask what you think about my suggestions. If you are fine with them feel free to merge :)

darioalessandro commented:
Roger that! Will review today

darioalessandro commented:
@finga looking good!!

finga commented:
> @finga looking good!!

Awesome, thx!!

Commit:
cargo: Remove `Cargo.lock` as this is a library

See [Cargo.toml
vs. Cargo.lock](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html)
and [Why do binaries have Cargo.lock in version control, but not
libraries?](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries)
for more information.

Commit:
cargo: Update versions of the lib and the example

Use only the major and minor version numbers and update the
dependencies of the example.

Commit:
cargo: Bump version to `1.0.0` (#5)

finga commented:
I took a look at the dependencies and took the liberty to update them. If you do not like how I did it just let me know and I can remove 3487229. Otherwise I think this should be ready to be merged.

darioalessandro commented:
Sounds good, I'll test locally and add a couple of github actions, then I'll go ahead and merge it

finga commented:
I do not want to be annoying, just wondering when I can count on having this released?

darioalessandro commented on .gitignore:
```diff
@@ -1,2 +1,3 @@
-target
\ No newline at end of file
+target
+.gitignore
```
adding gitignore to gitignore? I'll remove this entry

darioalessandro reviewed (approved):
lgtm

darioalessandro commented:
@finga It is out!!! https://crates.io/crates/yew-websocket 

Sorry for taking this long!

darioalessandro commented:
It is out @finga !!
